docs/src/proposals/append-vec-storage.md

```diff
@@ -0,0 +1,231 @@
+| lamport         | 7 bytes                                   |
++-------------------------------------------------------------+
+```
+
+In memory, each StoredAccountMetaV2 occupies 24 bytes.  The extra
+8 bytes is to store the derived `local_id` for accessing the pubkey
+entry and the `data_size` for loading account data.
+
+```
+pub StoredAccountMetaV2 {
+    pub local_id: u32, // derived when loading the AppendVec from the storage
+                       // the id can be used to fetch its pubkey
+    pub owner_local_id: u32,
+    data_offset: u32,
+    data_size: u32,   // derived when first loading the AppendVec from the storage
+                      // by taking the difference between its data_offset and the
+                      // data_offset of the next AccountMeta entry
+    status_and_lamport: u64,
+}
+```
+
+#### Account Pubkey Block
+The account pubkey is accessed via the `local_id` of the `StoredAccountMetaV2`,
+which is derived from the location of the account meta when an AppendVec is
+first opened.  Specifically, one account pubkey is located at
+`account_pubkey_block_offset` + 32 + `local_id` * `ACCOUNT_PUBKEY_SIZE`.
+
+```
++--------------------------------------+
+| Hash (32 bytes for the entire block) |
++--------------------------------------+
+| account pubkey 1                     |
++--------------------------------------+
+| account pubkey 2                     |
++--------------------------------------+
+|         ...                          |
++--------------------------------------+
+| account pubkey N                     |
++--------------------------------------+
+```
+
+```
++-------------------------------------------------------------+
+| Account Pubkey (32 bytes each)                              |
++-------------------------------------------------------------+
+| pubkey          | 32 bytes                                  |
++-------------------------------------------------------------+
+```
+
+#### Owners Block
+The owners block stores the owner's information.  An owner entry can be accessed
+via `owner_block_offset` + 32 + `owner_local_id` * `OWNER_ENTRY_SIZE`.
+
+```
++--------------------------------------+
+| Hash (32 bytes for the entire block) |
++--------------------------------------+
+| owner 1                              |
++--------------------------------------+
+| owner 2                              |
++--------------------------------------+
+|         ...                          |
++--------------------------------------+
+| owner N                              |
++--------------------------------------+
+```
+
+```
++-------------------------------------------------------------+
+| Owner (40 bytes each)                                       |
++-------------------------------------------------------------+
+| pubkey          | 32 bytes                                  |
+| write version   |  8 bytes                                  |
```

Comment:
We can get rid of write version if we know there is only 1 append vec per slot. These are all theoretical comments, not all the way thought through.

Reply:
> We can get rid of write version if we know there is only 1 append vec per slot.
Good to know this.  Looks like we can save 8 bytes per owner at least for now.

Btw, as you mentioned earlier that one AppendVec is currently capped at 16GB, can I know what will happen if there's a hot slot such that it goes beyond the 16GB limit?

Reply:
Offline discussed with Jeff.  Supporting multiple AppendVec per slot to enable multiple writers is not currently needed as we already have a write cache on top of AppendVec.